guvnor-ala/guvnor-ala-openshift/guvnor-ala-openshift-provider/src/main/java/org/guvnor/ala/openshift/access/OpenShiftTemplate.java

```diff
@@ -0,0 +1,179 @@
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreType;
+import io.fabric8.kubernetes.api.model.KubernetesList;
+import io.fabric8.kubernetes.client.dsl.Loadable;
+import io.fabric8.openshift.api.model.DoneableTemplate;
+import io.fabric8.openshift.api.model.Template;
+import io.fabric8.openshift.client.dsl.TemplateResource;
+import org.guvnor.ala.openshift.access.exceptions.OpenShiftClientException;
+import org.guvnor.ala.openshift.config.OpenShiftRuntimeConfig;
+
+/**
+ * Kubernetes template wrapper/utility for openshift ala.
+ */
+@JsonIgnoreType
+public class OpenShiftTemplate {
+
+    private final TemplateResource<Template, KubernetesList, DoneableTemplate> templateResource;
+
+    private OpenShiftTemplate(TemplateResource<Template, KubernetesList, DoneableTemplate> templateResource) {
+        this.templateResource = templateResource;
+    }
+
+    public Collection<Parameter> getParameters() {
+        return getParameterMap().values();
+    }
+
+    public SortedMap<String, Parameter> getParameterMap() {
+        SortedMap<String, Parameter> map = new TreeMap<String, Parameter>();
+        Template template = templateResource.get();
+        if (template != null) {
+            List<io.fabric8.openshift.api.model.Parameter> list = template.getParameters();
+            for (io.fabric8.openshift.api.model.Parameter param : list) {
+                map.put(param.getName(), new Parameter(param));
+            }
+        }
+        return map;
+    }
+
+    // package-protected only for use by OpenShiftClient
+    KubernetesList process(Map<String, String> parameters) {
+        return templateResource.process(parameters);
+    }
+
+    /**
+     * Gets an openshift client by runtime configuration.
+     * @param client the client
+     * @param runtimeConfig the runtime configuration
+     * @return the template
+     * @throws OpenShiftClientException
+     */
+    public static OpenShiftTemplate fromRuntimeConfig(OpenShiftClient client, OpenShiftRuntimeConfig runtimeConfig) {
+        String prjName = runtimeConfig.getProjectName();
+        String templateUri = runtimeConfig.getResourceTemplateUri();
+        String templateName = runtimeConfig.getResourceTemplateName();
+        OpenShiftTemplate template = from(client, runtimeConfig, prjName, templateUri, templateName);
+        if (template == null) {
+            throw new OpenShiftClientException(String.format("cannot load template with project [%s] and uri [%s] or name [%s]", prjName, templateUri, templateName));
+        }
+        return template;
+    }
+
+    /**
+     * Gets an openshift template by uri.
+     * @param client the client
+     * @param prjName the project/namespace
+     * @param templateUri the location of the template
+     * @return the template
+     * @throws OpenShiftClientException
+     */
+    public static OpenShiftTemplate fromUri(OpenShiftClient client, String prjName, String templateUri) {
```

Comment:
yes, I saw quickly the OpenShiftAccessInterfaceImpl().newOpenShiftClient() that creates the OpenShiftClient and for a moment I thought a client created this way took some system settings. 
But seems like not, ok, thanks.